src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.tt

```diff
@@ -169,32 +169,32 @@ namespace Microsoft.Data.Analysis
             switch (typeof(T))
             {
 <# foreach (TypeConfiguration type in typeConfiguration) { #>
-<# if (type.TypeName == "bool") { #>
+<# if (type.TypeName == "bool" || type.TypeName == "DateTime") { #>
                 case Type <#=type.TypeName#>Type when <#=type.TypeName#>Type == typeof(<#=type.TypeName#>):
 <# if (method.IsNumeric == true) { #>
                     throw new NotSupportedException();
 <# } else { #>
-                    if (typeof(U) != typeof(bool))
+                    if (typeof(U) != typeof(<#=type.TypeName#>))
                     {
                         throw new NotSupportedException();
                     }
 <# if (method.MethodType == MethodType.ComparisonScalar || method.MethodType == MethodType.Comparison) { #>
-                    PrimitiveDataFrameColumn<bool> retColumn = CloneAsBooleanColumn();
+                    PrimitiveDataFrameColumn<bool> ret<#=type.TypeName#>Column = CloneAsBooleanColumn();
```

Comment:
Should `CloneAsBooleanColumn` be `CloneAs<#=type.TypeName#>Column`

Reply:
I see the confusion! This is the result column, so it is a boolean column. We need to name the variable retDateTimeColumn because this code is duplicated for both boolean and date time, and the variables are in the same scope.